build-pyg-support.py

```diff
@@ -0,0 +1,235 @@
+#! /usr/bin/env python3
+
+import argparse
+import os
+import pathlib
+import shutil
+import subprocess
+import tempfile
+from collections import namedtuple
+from dataclasses import dataclass
+
+PackageMeta = namedtuple("PackageMeta", ["name", "tag", "url"])
+PACKAGES_META = {
+    "torch": PackageMeta("torch", "1.10.1", None),
+    "cluster": PackageMeta("pytorch-cluster", "*", "https://github.com/rusty1s/pytorch_cluster.git"),
+    "scatter": PackageMeta("pytorch-scatter", "2.0.8", "https://github.com/rusty1s/pytorch_scatter.git"),
+    "sparse": PackageMeta("pytorch-sparse", "0.6.12", "https://github.com/rusty1s/pytorch_sparse.git"),
+    "spline": PackageMeta("pytorch-spline-conv", "*", "https://github.com/rusty1s/pytorch_spline_conv.git"),
+    "pyg": PackageMeta("pyg", "2.0.1", "https://github.com/pyg-team/pytorch_geometric.git"),
+}
+
+
+def run_command(command, run=True):
+    print(" ".join(command))
+
+    if not run:
+        return
+
+    with subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE) as process:
+        while True:
+            output = process.stdout.readline()
+            if process.poll() is not None:
+                break
+            if output:
+                print(output.strip().decode("utf-8"))
+
+        _, stderr = process.communicate()
+
+        if process.returncode != 0:
+            raise RuntimeError(stderr.decode("utf-8"))
+
+
+def find_file(path, glob):
+    results = sorted(pathlib.Path(path).glob(str(glob)))
+
+    if len(results) > 1:
+        raise RuntimeError(f"Multiple matches for glob '{str(glob)}' in path '{str(path)}'.\n{str(results)}")
+
+    if len(results) < 1:
+        raise RuntimeError(f"No matches for glob '{str(glob)}' in path '{str(path)}'.")
+
+    return str(results[0].name)
```

Comment:
Why not just return the whole path? I don't see any reason to decemate the value to string.